### ISSUE TITLE
Deleting a template group's index results in a confirmation but also fails

### ISSUE THREAD

**Issue by murtaugh** [ExpressionEngine/ExpressionEngine#532]
In EE 5.3.2 I tried to delete the index template of a template group. I got a message that it was successful, but it still existed.

**How To Reproduce**

1. Create template group.
2. Delete the index template.
3. You get a confirmation message, but will fail.

<img width="443" alt="Screen Shot 2020-09-25 at 2 34 52 PM" src="https://user-images.githubusercontent.com/308752/94303831-76888300-ff3c-11ea-9ca5-37e8e9057e9e.png">

I actually presume this is the intended behavior, in which case the message should be an error instead of a confirmation.

**Comment by intoeetive** [comment]
The template is actually being deleted, it's just being recreated right after (if you check the contents, it should be empty) - and this happens if you have "set templates as files" setting turned on.

So as such, the message is correct

Do you think there needs to be another message saying the template has been automatically created? (I'd say no)

**Comment by murtaugh** [comment]
Makes sense, but I think the fact that index pages have a special status that isn't reflected in the UI results in confusion.

Here are some thoughts, some good, some maybe not:

- When editing an index template, place a small note on the Edit Template page informing users about the nature of index templates.
- Make index templates un-deletable via the CP. Users can still clear them out if they want.
- If a user tries to delete the file and it reappears, perhaps put a comment in the file itself explaining the continued existence of the file.
- Make empty/missing index files redirect to the home page.
